apis/condition_set.go

```diff
@@ -52,6 +52,17 @@ type ConditionManager interface {
 	// set to true.
 	IsHappy() bool
 
+	// IsUnknown looks at the happy condition and returns true if that condition is
+	// set to unknown..
+	IsUnknown() bool
+
+	// IsFalse looks at the happy condition and returns true if that condition is
+	// set to false.
+	IsFalse() bool
```

Comment:
Can we just drop these two functions? Instead, a caller can just call `GetHappeyCondition().IsUnknown()`.

Reply:
I thought the goal of ConditionManager is to make it easier for users to operate on resources' conditions and it prevents users from dealing with the business logic of underlying `Condition`.  
By following the logic you mentioned, should we also deprecate `IsHappy() bool` for ConditionManager? In addition, `GetHappeyCondition()` can also be dropped since we can use `GetCondition(t ConditionType) *Condition` to get the happy condition. I think whether we can add these two functions into ConditionManger should depend on what ConditionManager aims to provide.

Reply:
The goal of the condition manager is to allow the manager to manage conditions on behalf of the controller.

`.happy` is used internally to understand which condition to propagate updates to. Externally the class already knows which is happy (which is a very silly name and should not be leaked more than it is) because it set from consts in the controlling class.